tests/core/v5_1/test_network.py

```diff
@@ -232,3 +234,53 @@ async def _do_talk_response(network):
                     )
 
     assert response == b"test-response-payload"
+
+
+class ProtocolA(TalkProtocolAPI):
+    protocol_id = b"a"
+
+
+class ProtocolB(TalkProtocolAPI):
+    protocol_id = b"b"
+
+
+@pytest.mark.trio
+async def test_network_add_talk_protocol(alice):
+    async with alice.network() as alice_network:
+        alice_network.add_talk_protocol(ProtocolA())
+
+        with pytest.raises(DuplicateProtocol):
+            alice_network.add_talk_protocol(ProtocolA())
+
+        alice_network.add_talk_protocol(ProtocolB())
+
+        with pytest.raises(DuplicateProtocol):
+            alice_network.add_talk_protocol(ProtocolB())
+
+
+@pytest.mark.trio
+async def test_network_responds_to_unhandled_protocol_messages(
+    alice, bob, autojump_clock
+):
+    bob.enr_db.set_enr(alice.enr)
+
+    async with alice.network() as alice_network:
+        async with bob.network() as bob_network:
+            with trio.fail_after(1):
+                response = await bob_network.talk(
+                    node_id=alice.node_id,
+                    protocol=ProtocolA.protocol_id,
+                    payload=b"payload-a",
+                    endpoint=alice.endpoint,
+                )
+            assert response == b""
+
+            alice_network.add_talk_protocol(ProtocolA())
+
+            with pytest.raises(trio.EndOfChannel):
```

Comment:
Why should this return an `EndOfChannel`?

Reply:
Yeah, it's probably time for this to get addressed in the base protocol.  If you recall, you recently identified that I had duplicate timeouts being enforced on request/response.  In removing the outer timeout, the inner-timeout now manifests as a `trio.EndOfChannel`.  This is because request/response is implemented using a channel that the response is fed through.  When the inner timeout is triggered, the channel gets closed, which causes the part listening for the response to raise `trio.EndOfChannel`.

I think that we probably want to catch this internally and re-raise it as a `trio.TooSlowError`.